File: sem3/map/Lab/2/src/model/IfStmt.java
package model;

public class IfStmt implements IStmt {
    private Exp exp;
    private IStmt thenS;
    private IStmt elseS;

    public IfStmt(Exp exp, IStmt thenS, IStmt elseS) {
        this.exp = exp;
        this.thenS = thenS;
        this.elseS = elseS;
    }

    @Override
    public String toString() {
        return "if(" + exp.toString() + ") then " + thenS.toString() + " else " + elseS.toString() + " end";
    }

    @Override
    public PrgState execute(PrgState state) {
        if(exp.eval(state.getSymTable()) == 0)
            state.getExeStack().push(elseS);
        else
            state.getExeStack().push(thenS);
        return null;
    }
}

File: sem3/sem 3/~craciunf/c-sharp-toy-language-interpreter/models/IfStmt.cs
/* package models; */
using System;
using System.Collections.Generic;

namespace ToyLanguageInterpreter {
    public class IfStmt : IStmt {
        private Exp exp;
        private IStmt thenStmt, elseStmt;

        public IfStmt(Exp exp, IStmt thenStmt, IStmt elseStmt) {
            this.exp = exp;
            this.thenStmt = thenStmt;
            this.elseStmt = elseStmt;
        }

        /* @Override */
        public String toString() {
            return "if(" + this.exp.toString() + ") then(" + this.thenStmt.toString() + ") else(" + this.elseStmt.toString() + ")";
        }

        /* @Override */
        public PrgState execute(PrgState state) {
            Stack <IStmt> exeStack = state.getExeStack();
            Dictionary <String, int> symTable = state.getSymTable();

            if(this.exp.eval(symTable) != 0)
                exeStack.Push(thenStmt);
            else
                exeStack.Push(elseStmt);

            return state;
        }
    }
}
